packages/rum-core/src/boot/startRum.ts

```diff
@@ -48,6 +48,23 @@ export function startRum(
       { view: { name: null } }
     )
   )
+  internalMonitoring.setTelemetryContextProvider(() => ({
+    application: {
+      id: configuration.applicationId,
+    },
+    session: {
+      id: session.findTrackedSession()?.id,
+    },
+    view: {
+      id: parentContexts.findView()?.id,
+    },
+    action: {
+      id: parentContexts.findAction()?.id,
+    },
+  }))
+  internalMonitoring.telemetryEventObservable.subscribe((event) => {
+    lifeCycle.notify(LifeCycleEventType.TELEMETRY_EVENT_COLLECTED, event)
+  })
```

Comment:
I though about it but:
- it was an easier change since rum batch is in rum event collection so it would add a new dependency to a couple of methods signatures and their corresponding tests
- it is more consistent with how we deal with RUM_EVENT_COLLECTED 

any reason to push the other way?

Reply:
As I see it, adding complexity to the LifeCycle adds complexity to the whole RUM SDK since it is a central piece used everywhere. I understand that it's a good way to decouple some modules to ease unit tests.

But here we have a chance to not use the LifeCycle and keep the complexity local instead. It would be one less indirection, making things easier to follow. Also it would be more consistent with Logs.